llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -6855,6 +6855,52 @@ Instruction *InstCombinerImpl::visitICmpInst(ICmpInst &I) {
   if (Value *V = simplifyICmpInst(I.getPredicate(), Op0, Op1, Q))
     return replaceInstUsesWith(I, V);
 
+  {
+    Value *X;
+    const APInt *C1, *C2;
+    ICmpInst::Predicate Pred;
+    if ((match(&I, m_ICmp(Pred, m_SRem(m_Value(X), m_NonNegative(C1)),
+                          m_APInt(C2))) &&
+         ((Pred == ICmpInst::ICMP_SLT && *C2 == *C1 - 1) ||
+          (Pred == ICmpInst::ICMP_SGT && *C2 == *C1 - 2))) ||
+        (match(&I,
+               m_ICmp(Pred, m_SRem(m_Value(X), m_Negative(C1)), m_APInt(C2))) &&
+         *C2 == *C1 + 1 &&
+         (Pred == ICmpInst::ICMP_SGT || Pred == ICmpInst::ICMP_SLE))) {
+      // icmp slt (X s% C), (C - 1) --> icmp ne (X s% C), (C - 1), if C >= 0
+      // icmp sgt (X s% C), (C - 2) --> icmp eq (X s% C), (C - 1), if C >= 0
+      // icmp sgt (X s% C), (C + 1) --> icmp ne (X s% C), (C + 1), if C < 0
+      // icmp sle (X s% C), (C + 1) --> icmp eq (X s% C), (C + 1), if C < 0
+      return CmpInst::Create(
+          Instruction::ICmp,
+          (Pred == ICmpInst::ICMP_SLT ||
+           (Pred == ICmpInst::ICMP_SGT && *C2 == *C1 - 1))
+              ? ICmpInst::ICMP_NE
+              : ICmpInst::ICMP_EQ,
+          Op0,
+          Constant::getIntegerValue(Op1->getType(), *C2 + (*C2 == *C1 - 2)));
```

Comment:
```suggestion
          ConstantInt::get(Op1->getType(), *C2 + (*C2 == *C1 - 2)));
```
Use `ConstantInt::get` instead. TBH they are equivalent but `ConstantInt::get` is more common.